src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -63,10 +63,13 @@ export default App.extend({
     'change': 'onChangeState',
   },
   onChangeState(state) {
-    if (state.hasChanged('groupId')) {
+    if (state.hasChanged('groupId') || state.hasChanged('worklistType')) {
```

Comment:
I don't think we should store the type on the state.  we're going to want to store the state model in the future for recalling previous sort/filter and so the type really only needs to come from the url and can probably just be an option like the id